diff --git a/myosuite/envs/myo/myowheelchair/ParallelTrainingScriptWandB.py b/myosuite/envs/myo/myowheelchair/ParallelTrainingScriptWandB.py
index 10797ef..d219379 100644
--- a/myosuite/envs/myo/myowheelchair/ParallelTrainingScriptWandB.py
+++ b/myosuite/envs/myo/myowheelchair/ParallelTrainingScriptWandB.py
@@ -162,17 +162,17 @@ if __name__ == "__main__":
 
     print("Begin training")
 
+    model = PPO('MlpPolicy', envs, verbose=1, ent_coef=0.001, policy_kwargs=policy_kwargs)
+    
     # TODO TRY LOADING
-    # model = PPO('MlpPolicy', envs, verbose=1, ent_coef=0.001, policy_kwargs=policy_kwargs)
-
-    model_num =   '2025_07_16_13_12_43' # Loaded WheelDist_policy_GOOD
-    model = PPO.load('./MPL_baselines/policy_best_model'+ '/'+ env_name + '/' + model_num + r'/best_model', envs, verbose = 1, ent_coeff = 0.01, policy_kwargs = policy_kwargs)
+    # model_num =   '2025_07_16_13_12_43' # Loaded WheelDist_policy_GOOD
+    # model = PPO.load('./MPL_baselines/policy_best_model'+ '/'+ env_name + '/' + model_num + r'/best_model', envs, verbose = 1, ent_coeff = 0.01, policy_kwargs = policy_kwargs)
 
     obs_callback = TensorboardCallback()
     callback = CallbackList([eval_callback, WandbCallback(gradient_save_freq=100)])#, obs_callback])
 
     #TODO TOTAL TIMESTEPS HERE
-    model.learn(total_timesteps=1000, tb_log_name=env_name + "_" + time_now, callback=callback)
+    model.learn(total_timesteps=5e5, tb_log_name=env_name + "_" + time_now, callback=callback)
     model.save(curr_dir+'/WheelDist_policy')
 
     # Record video after training
diff --git a/myosuite/envs/myo/myowheelchair/WheelDist_policy.zip b/myosuite/envs/myo/myowheelchair/WheelDist_policy.zip
index e12cca7..cf68f72 100644
Binary files a/myosuite/envs/myo/myowheelchair/WheelDist_policy.zip and b/myosuite/envs/myo/myowheelchair/WheelDist_policy.zip differ
diff --git a/myosuite/envs/myo/myowheelchair/render.py b/myosuite/envs/myo/myowheelchair/render.py
index 8045fdf..797577d 100644
--- a/myosuite/envs/myo/myowheelchair/render.py
+++ b/myosuite/envs/myo/myowheelchair/render.py
@@ -19,7 +19,7 @@ if __name__ == "__main__":
     # render
     frames = []
     for _ in range(300):
-        frames.append(env.sim.renderer.render_offscreen(width=400, height=400, camera_id=4)) 
+        frames.append(env.sim.renderer.render_offscreen(width=400, height=400, camera_id=5)) 
         o = env.get_obs()
         a = pi.predict(o)[0]
         next_o, r, done, *_, ifo = env.step(
diff --git a/myosuite/envs/myo/myowheelchair/videos/HandFocusedRender.mp4 b/myosuite/envs/myo/myowheelchair/videos/HandFocusedRender.mp4
index 364e26e..b395191 100644
Binary files a/myosuite/envs/myo/myowheelchair/videos/HandFocusedRender.mp4 and b/myosuite/envs/myo/myowheelchair/videos/HandFocusedRender.mp4 differ
diff --git a/myosuite/envs/myo/myowheelchair/videos/Wheel_MinDist.mp4 b/myosuite/envs/myo/myowheelchair/videos/Wheel_MinDist.mp4
index 13c3592..af74647 100644
Binary files a/myosuite/envs/myo/myowheelchair/videos/Wheel_MinDist.mp4 and b/myosuite/envs/myo/myowheelchair/videos/Wheel_MinDist.mp4 differ
diff --git a/myosuite/envs/myo/myowheelchair/wheelhold_v0.py b/myosuite/envs/myo/myowheelchair/wheelhold_v0.py
index a3c6174..e160acb 100644
--- a/myosuite/envs/myo/myowheelchair/wheelhold_v0.py
+++ b/myosuite/envs/myo/myowheelchair/wheelhold_v0.py
@@ -15,9 +15,9 @@ class WheelHoldFixedEnvV0(BaseV0):
 
     DEFAULT_OBS_KEYS = ['time', 'wheel_err_right', 'hand_qpos', 'hand_qvel']
     DEFAULT_RWD_KEYS_AND_WEIGHTS = {
-        "goal_dist": 150.0,
-        "hand_dist" : 50.0,
-        "fin_open": -15.0,
+        "goal_dist": 50.0,
+        "hand_dist" : 25.0,
+        "fin_open": -100.0,
         "bonus": 0.0,
         "penalty": 20,
     }
@@ -125,7 +125,7 @@ class WheelHoldFixedEnvV0(BaseV0):
         hand_initpos_err_right = np.linalg.norm(obs_dict['hand_initpos_err_right'])
         
         act_mag = np.linalg.norm(self.obs_dict['act'], axis=-1)/self.sim.model.na if self.sim.model.na !=0 else 0
-        drop = dist_right > 0.300
+        drop = dist_right > 0.500
 
         fin_keys = ['fin0', 'fin1', 'fin2', 'fin3', 'fin4']
         # for fin in fin_keys:
@@ -142,18 +142,18 @@ class WheelHoldFixedEnvV0(BaseV0):
 
         rwd_dict = collections.OrderedDict((
             ('goal_dist', math.exp(-2.0*abs(dist_right))), #exp(- k * abs(x))
-            ('hand_dist', math.exp(-2.0*abs(hand_initpos_err_right))),
+            ('hand_dist', math.exp(-1.0*abs(hand_initpos_err_right))),
             ('bonus', 1.*(dist_right<2*0) + 1.*(dist_right<0)),
             ('act_reg', -1.*act_mag),
-            ("fin_open", np.exp(-5 * fin_open)),  # fin_open + np.log(fin_open +1e-8)
+            ("fin_open", np.exp(-10 * fin_open)),  # fin_open + np.log(fin_open +1e-8)
 
             #('grip_bonus', 1.0 * grip_right),
             ('penalty', -1.*drop),
             ('sparse', dist_right < 0.055),
             #('sparse', 1.0 * grip_right - dist_right),
-            ('solved', dist_right < 0.015),
+            ('solved', dist_right < 0.001),
             #('solved', grip_right and dist_right < 0.015),
-            ('done', dist_right > 0.515),
+            ('done', dist_right > 0.9),
         ))
         
         rwd_dict['dense'] = np.sum([wt*rwd_dict[key] for key, wt in self.rwd_keys_wt.items()], axis=0)
Submodule myosuite/simhive/myo_sim 3b866e0..587629a:
diff --git a/myosuite/simhive/myo_sim/wheelchair/assets/myowheelchair_chain.xml b/myosuite/simhive/myo_sim/wheelchair/assets/myowheelchair_chain.xml
index d0205cb..e41adb8 100644
--- a/myosuite/simhive/myo_sim/wheelchair/assets/myowheelchair_chain.xml
+++ b/myosuite/simhive/myo_sim/wheelchair/assets/myowheelchair_chain.xml
@@ -41,6 +41,7 @@
             <site name="wheelchair_grip_left" pos="-0.0371 0.27 0.05" size="0.005" rgba="0 1 0 0.5"/>
         </body>
         <site name="hand_start_right" pos="0.5324 -0.3142 -0.1024" size="0.008" rgba="1 0 0 1"/>
+        <site name="rail_bottom_right" pos="0.5324 -0.33 -0.1024" size="0.005" rgba="0 0 1 1"/>
         <body name = "right_rear_wheel" pos="0.4953 -0.5842 -0.1524">
             <joint name= "right_rear" axis="-1 0 0" damping=".03"/>
             <geom mesh= "rear_wheel2" pos="-0.4953 0.5842 0.1524" material="dark_grey" conaffinity="0" contype="0"/>
@@ -64,6 +65,7 @@
             <geom mesh="handrail_coll16" pos="0.0371 0 0" rgba="0 0 1 0"/>
             
             <site name="wheelchair_grip_right" pos="0.0371 0.27 0.05" size="0.005" rgba="0 1 0 1"/>
+
             <!-- <geom type = "sphere" pos="0.0371 0.27 0.05" size = "0.008" rgba="1 0 0 1"/> -->
         </body>
         
diff --git a/myosuite/simhive/myo_sim/wheelchairandhuman/Wheelchairwitharm/myoarm_assets_for_wc.xml b/myosuite/simhive/myo_sim/wheelchairandhuman/Wheelchairwitharm/myoarm_assets_for_wc.xml
index e078871..d8a8702 100644
--- a/myosuite/simhive/myo_sim/wheelchairandhuman/Wheelchairwitharm/myoarm_assets_for_wc.xml
+++ b/myosuite/simhive/myo_sim/wheelchairandhuman/Wheelchairwitharm/myoarm_assets_for_wc.xml
@@ -80,7 +80,6 @@
         <mesh file="../myo_sim/meshes/scaphoid.stl" name="proximal_row_geom_2_sdfastSCAPHOIDw"/>
         <mesh file="../myo_sim/meshes/pisiform.stl" name="proximal_row_geom_3_sdfastPISIFORMw"/>
         <mesh file="../myo_sim/meshes/triquetrum.stl" name="proximal_row_geom_4_sdfastTRIQUETRALw"/>
-
         <mesh name="thorax" file="../myo_sim/meshes/thorax.stl" scale="1 1 1"/>
         <mesh name="clavicle" file="../myo_sim/meshes/clavicle.stl" scale="1 1 1"/>
         <mesh name="scapula" file="../myo_sim/meshes/scapula.stl" scale="1 1 1"/>
diff --git a/myosuite/simhive/myo_sim/wheelchairandhuman/Wheelchairwitharm/myotorso_abdomen_chain+arm.xml b/myosuite/simhive/myo_sim/wheelchairandhuman/Wheelchairwitharm/myotorso_abdomen_chain+arm.xml
index 86b728f..26b1a75 100644
--- a/myosuite/simhive/myo_sim/wheelchairandhuman/Wheelchairwitharm/myotorso_abdomen_chain+arm.xml
+++ b/myosuite/simhive/myo_sim/wheelchairandhuman/Wheelchairwitharm/myotorso_abdomen_chain+arm.xml
@@ -37,7 +37,7 @@
             <geom mesh="torso_geom_1_hat_spine" name="torso_geom_1" type="mesh"/>
             <geom mesh="torso_geom_2_hat_jaw" name="torso_geom_2" type="mesh"/>
             <geom mesh="torso_geom_3_hat_skull" name="torso_geom_3" type="mesh"/>
-            <geom mesh="torso_geom_4_hat_ribs" name="torso_geom_4" type="mesh"/>
+            <geom mesh="torso_geom_4_hat_ribs_no_scapcla" name="torso_geom_4_hat_ribs_no_scapcla" type="mesh"/>
 
             <site name="ercspn_r_ercspn_r-P2" pos="-0.055 0.11 0.0241"/>
             <site name="ercspn_l_ercspn_l-P2" pos="-0.055 0.11 -0.0241"/>
@@ -53,7 +53,7 @@
             <site name="spine_1" pos="0.030 0 0" class="myotorso_marker"/>
             <site name="thoracis_spine" pos="-.018 .194 0" class="myotorso_marker"/>
             <site name="chest" pos="0.084 0.221 0" class="myotorso_marker"/>
-            <body name="arm_pos" pos="0.038 0.374 0.024" euler = "-1.57 0 1.57" >
+            <body name="arm_pos" pos="-0.01 0.376 0.025" euler = "-1.57 -0.1 1.57" >
                 <include file="../../../../simhive/myo_sim/wheelchairandhuman/Wheelchairwitharm/myoarm_body_for_wc.xml"/>
             </body>
         </body>
diff --git a/myosuite/simhive/myo_sim/wheelchairandhuman/assets/myotorso_abdomen_assets.xml b/myosuite/simhive/myo_sim/wheelchairandhuman/assets/myotorso_abdomen_assets.xml
index 477e618..2326915 100644
--- a/myosuite/simhive/myo_sim/wheelchairandhuman/assets/myotorso_abdomen_assets.xml
+++ b/myosuite/simhive/myo_sim/wheelchairandhuman/assets/myotorso_abdomen_assets.xml
@@ -39,7 +39,7 @@
         <mesh file="../myo_sim/meshes/hat_spine.stl" name="torso_geom_1_hat_spine"/>
         <mesh file="../myo_sim/meshes/hat_jaw.stl" name="torso_geom_2_hat_jaw"/>
         <mesh file="../myo_sim/meshes/hat_skull.stl" name="torso_geom_3_hat_skull"/>
-        <mesh file="../myo_sim/meshes/hat_ribs_scap.stl" name="torso_geom_4_hat_ribs"/>
+        <mesh file="../myo_sim/meshes/hat_ribs_scap_no_scapcla.stl" name="torso_geom_4_hat_ribs_no_scapcla"/>
 
         <material name="MatSkin_myotorso" specular="0.2" shininess=".1" reflectance="0.5" rgba="0.9 .7 1 1"/>
     </asset>
